File: backend/src/TalentVault.Domain/Entities/Candidate.cs
namespace TalentVault.Domain.Entities;

public class Candidate
{
    public Guid Id { get; set; }
    public Guid CompanyId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string State { get; set; } = string.Empty;
    public string Seniority { get; set; } = string.Empty;
    public string? ResumeUrl { get; set; }
    public DateTime CreatedAt { get; set; }

    // Navigation properties
    public Company Company { get; set; } = null!;
    public ICollection<CandidateSkill> Skills { get; set; } = new List<CandidateSkill>();
}

File: front/types/candidate.ts
export interface Candidate {
  id: string;
  name: string;
  email: string;
  phone: string;
  city: string;
  state: string;
  seniority: string;
  resumeUrl?: string | null;
  createdAt: string;
}

export interface CandidatesPageResponse {
  items: Candidate[];
  page: number;
  pageSize: number;
  total: number;
}

export interface CandidateFilters {
  city?: string;
  seniority?: string;
  skills?: string;
}

export interface ApiResponse<T> {
  success: boolean;
  data: T;
  errors: string[];
}

export interface CreateCandidateRequest {
  name: string;
  email: string;
  phone: string;
  city: string;
  state?: string;
}

export interface PublicCandidateSubmissionRequest extends CreateCandidateRequest {
  file: File;
}
